app/views/questionnaire.py

```diff
@@ -738,15 +738,20 @@ def _build_template(current_location, context, template, schema, answer_store, m
     previous_location = path_finder.get_previous_location(current_location)
     previous_url = previous_location.url(metadata) if previous_location is not None else None
 
-    return _render_template(context, current_location, template, front_end_navigation, previous_url, schema, metadata, answer_store)
+    add_person_url = None
+    if routing_path:
+        add_person_location = routing_path[routing_path.index(current_location) - 1]
+        add_person_url = add_person_location.url(metadata) if add_person_location is not None else None
```

Comment:
Do we need `is not None`?

Reply:
Not really, I copied the previous_url line and modified accordingly.